lib/files.js

```diff
@@ -10,14 +10,24 @@ var path = require('path');
 var options = require('./args.js');
 var log = require('davlog');
 var verify = require('./verify.js');
+var hooks = require('./hooks');
 
 var putBall = function(info, callback) {
     var localBall = path.join(options.dir, info.path);
-    process.nextTick(function() {
-        mkdirp(path.dirname(localBall), function() {
-            process.nextTick(function() {
-                info.tarball = localBall;
-                verify.verify(info, callback);
+    info.tarball = localBall;
+
+    hooks.tarball(info, callback, function () {
```

Comment:
[In `hooks.js`](https://github.com/bengl/registry-static/blob/212433f4584fe5f65fe2479e70ed5ca698d7518c/lib/hooks.js#L30-L40), the functions are wrapped so that there's a "don't save"/bypass callback and "go ahead and do it" callback. This simplifies calling them throughout the codebase, so it's only a one-liner instead of doing the `shouldSave` logic every time in-line.